### PR TITLE
Add tikv-br doc

### PR DESCRIPTION
**Pull request by haojinming**
Signed-off-by: haojinming <jinming.hao@pingcap.com>

### What is changed?
Add RawKV BR description doc in dev features.
Almost same with the [tikv-br repo](https://github.com/tikv/migration/tree/main/br)

Preview [here](https://deploy-preview-325--tikv.netlify.app/docs/dev/concepts/explore-tikv-features/backup-restore/).

### Which version does your change affect?
dev

### CODE REVIEW

**Comment by haojinming** [comment]
@pingyu @zeminzhou PTAL~ thanks.

File: content/docs/dev/concepts/explore-tikv-features/backup-restore.md [hunk 1]
```diff
@@ -10,12 +10,156 @@ menu:
...
+```
+[2022/09/20 18:01:10.125 +08:00] [INFO] [collector.go:67] ["Raw backup success summary"] [total-ranges=3] [ranges-succeed=3] [ranges-failed=0] [backup-total-regions=3] [total-take=5.050265883s] [backup-ts=436120585518448641] [total-kv=100000] [total-kv-size=108.7MB] [average-speed=21.11MB/s] [backup-data-size(after-compressed)=78.3MB]
+```
+Explanations for the above message are as follows: 
+- `total-ranges`: Number of ranges that the whole backup task is split to. Equals to `ranges-succeed` + `ranges-failed`.
+- `ranges-succeed`: Number of succeeded ranges.
+- `ranges-failed`: Number of failed ranges.
+- `backup-total-regions`: The tikv regions that backup takes.
+- `total-take`: The backup duration.
+- `backup-ts`: The backup start timestamp, only take effect for APIV2 TiKV cluster, which can be used as `start-ts` of `TiKV-CDC` when creating replication tasks. Refer to [Create a replication task](https://github.com/tikv/migration/blob/main/cdc/README.md#create-a-replication-task).
```
Comment (pingyu) [inline content/docs/dev/concepts/explore-tikv-features/backup-restore.md:104]:
```suggestion
- `backup-ts`: The backup start timestamp, only take effect for API V2 TiKV cluster, which can be used as `start-ts` of `TiKV-CDC` when creating replication tasks. Refer to [Create a replication task](https://github.com/tikv/migration/blob/main/cdc/README.md#create-a-replication-task).
```

File: content/docs/dev/concepts/explore-tikv-features/backup-restore.md [hunk 2]
```diff
@@ -10,12 +10,156 @@ menu:
...
+- `total-kv`: Total kv count in backup files.
```
Comment (pingyu) [inline content/docs/dev/concepts/explore-tikv-features/backup-restore.md:105]:
```suggestion
- `total-kv`: Total number of key-value pairs in backup files.
```

File: content/docs/dev/concepts/explore-tikv-features/backup-restore.md [hunk 3]
```diff
@@ -10,12 +10,156 @@ menu:
...
+- `total-kv-size`: Total kv size in backup files. Note that this is the original size before compression.
```
Comment (pingyu) [inline content/docs/dev/concepts/explore-tikv-features/backup-restore.md:106]:
```suggestion
- `total-kv-size`: Total size of key-value pairs in backup files. Note that this is the original size before compression.
```

File: content/docs/dev/concepts/explore-tikv-features/backup-restore.md [hunk 4]
```diff
@@ -10,12 +10,156 @@ menu:
...
+tikv-br restore raw \
+    --pd "${PDIP}:2379" \
+    --storage "s3://backup-data/2022-09-16/" \
+    --ratelimit 128 \
+    --log-file restoreraw.log
+```
+Explanations for some options in the above command are as follows:
+
+- `--ratelimit`: The maximum speed at which a restoration operation is performed (MiB/s) on each `TiKV` node.
+- `--log-file`: Writing the TiKV-BR log to the `restorefull.log` file.
```
Comment (pingyu) [inline content/docs/dev/concepts/explore-tikv-features/backup-restore.md:127]:
```suggestion
- `--log-file`: Writing the TiKV-BR log to the `restoreraw.log` file.
```

File: content/docs/dev/concepts/explore-tikv-features/backup-restore.md [hunk 5]
```diff
@@ -10,12 +10,156 @@ menu:
...
+Restore Raw <---------/...............................................> 17.12%.
+```
+
+After restoration finish, the result message is displayed as follows:
+```
+[2022/09/20 18:02:12.540 +08:00] [INFO] [collector.go:67] ["Raw restore success summary"] [total-ranges=3] [ranges-succeed=3] [ranges-failed=0] [restore-files=3] [total-take=950.460846ms] [restore-data-size(after-compressed)=78.3MB] [total-kv=100000] [total-kv-size=108.7MB] [average-speed=114.4MB/s]
+```
+Explanations for the above message are as follows: 
+- `total-ranges`: Number of ranges that the whole backup task is split to. Equals to `ranges-succeed` + `ranges-failed`.
+- `ranges-succeed`: Number of succeeded ranges.
```
Comment (pingyu) [inline content/docs/dev/concepts/explore-tikv-features/backup-restore.md:145]:
```suggestion
- `ranges-succeed`: Number of successful ranges.
```

File: content/docs/dev/concepts/explore-tikv-features/backup-restore.md [hunk 6]
```diff
@@ -10,12 +10,156 @@ menu:
...
+- `ranges-failed`: Number of failed ranges.
+- `restore-files`: Number of restored files.
+- `total-take`: The restoration duration.
+- `total-kv`: Total restored kv count.
```
Comment (pingyu) [inline content/docs/dev/concepts/explore-tikv-features/backup-restore.md:149]:
```suggestion
- `total-kv`: Total number of restored key-value pairs.
```

File: content/docs/dev/concepts/explore-tikv-features/backup-restore.md [hunk 7]
```diff
@@ -10,12 +10,156 @@ menu:
...
+- `total-kv-size`: Total restored kv size. Note that this is the original size before compression.
```
Comment (pingyu) [inline content/docs/dev/concepts/explore-tikv-features/backup-restore.md:150]:
```suggestion
- `total-kv-size`: Total size restored key-value pairs. Note that this is the original size before compression.
```

File: content/docs/dev/concepts/explore-tikv-features/backup-restore.md [hunk 8]
```diff
@@ -10,12 +10,156 @@ menu:
...
+
+
+### Data Verification of Backup & Restore
+
+TiKV-BR can do checksum between TiKV cluster and backup files after backup or restoration finish with the config `--checksum=true`. Checksum is using the [checksum](https://github.com/tikv/client-go/blob/ffaaf7131a8df6ab4e858bf27e39cd7445cf7929/rawkv/rawkv.go#L584) interface in TiKV [client-go](https://github.com/tikv/client-go), which send checksum request to all TiKV regions to calculate the checksum of all **VALID** data. Then compare to the checksum value of backup files which is calculated during backup process.
+
+In some scenario, data is stored in TiKV with [TTL](https://docs.pingcap.com/tidb/stable/tikv-configuration-file#enable-ttl). If data is expired during backup & restore, the persisted checksum in backup files is different from the checksum of TiKV cluster. So checksum should not enabled in this scenario. User can perform a full comparison for all existing non-expired data between backup cluster and restore cluster with [scan](https://github.com/tikv/client-go/blob/ffaaf7131a8df6ab4e858bf27e39cd7445cf7929/rawkv/rawkv.go#L492) interface in [client-go](https://github.com/tikv/client-go).
```
Comment (pingyu) [inline content/docs/dev/concepts/explore-tikv-features/backup-restore.md:159]:
```suggestion
Please note that if data is stored in TiKV with [TTL](./ttl), and expiration happens during backup or restore, the persisted checksum in backup files must be different from the checksum of TiKV cluster. So checksum should **NOT** be enabled in this scenario. To verify correctness of backup and restoration in this scenario, you can perform a full comparison for all existing non-expired data between backup cluster and restore cluster by using [scan](../../develop/rawkv/scan) interface.
```

File: content/docs/dev/concepts/explore-tikv-features/backup-restore.md [hunk 9]
```diff
@@ -10,12 +10,156 @@ menu:
...
+- `total-take`: The restoration duration.
+- `total-kv`: Total restored kv count.
+- `total-kv-size`: Total restored kv size. Note that this is the original size before compression.
+- `average-speed`: The restoration speed, which approximately equals to `total-kv-size` / `total-take`.
+- `restore-data-size(after-compressed)`: The restoration file size.
+
+
+### Data Verification of Backup & Restore
+
+TiKV-BR can do checksum between TiKV cluster and backup files after backup or restoration finish with the config `--checksum=true`. Checksum is using the [checksum](https://github.com/tikv/client-go/blob/ffaaf7131a8df6ab4e858bf27e39cd7445cf7929/rawkv/rawkv.go#L584) interface in TiKV [client-go](https://github.com/tikv/client-go), which send checksum request to all TiKV regions to calculate the checksum of all **VALID** data. Then compare to the checksum value of backup files which is calculated during backup process.
```
Comment (pingyu) [inline content/docs/dev/concepts/explore-tikv-features/backup-restore.md:157]: TODO: change link of `checksum` after we add a `checksum` introduction [here](https://deploy-preview-325--tikv.netlify.app/docs/dev/develop/rawkv/introduction/).

**Review by pingyu — APPROVED** [review]
LGTM~

**Comment by haojinming** [comment]
@sunxiaoguang  @shichun-0415 PTAL~ thanks

File: content/docs/dev/concepts/explore-tikv-features/backup-restore.md [hunk 10]
```diff
@@ -10,12 +10,156 @@ menu:
...
+    caption="TiKV-BR architecture"
+    number="1" >}}
 
-## Usage Demonstration
 
-*TODO*
+### Recommended Deployment Configuration
+- In production environments, deploy `TiKV-BR` on a node with at least 8 cores CPU and 16 GB memory. Select an appropriate OS version by following [Linux OS version requirements](https://docs.pingcap.com/tidb/dev/hardware-and-software-requirements#linux-os-version-requirements).
+
+- Save backup data to Amazon S3 or one mounted network disk on all `TiKV-BR` and `TiKV` nodes.
```
Comment (sunxiaoguang) [inline content/docs/dev/concepts/explore-tikv-features/backup-restore.md:26]:
```suggestion
- Save backup data to Amazon S3 or other shared storage, for example mounting a NFS on all `TiKV-BR` and `TiKV` nodes.
```

File: content/docs/dev/concepts/explore-tikv-features/backup-restore.md [hunk 11]
```diff
@@ -10,12 +10,156 @@ menu:
...
+
+- Allocate sufficient resources for backup and restoration.
+
+- TiKV-BR only support raw data backup/restoration in TiKV cluster with version >= `5.0.0`.
+
+TiKV-BR, TiKV nodes, and the backup storage system should provide network bandwidth that is greater than the backup speed. If the target cluster is particularly large, the threshold of backup and restoration speed is limited by the bandwidth of the backup network.
```
Comment (sunxiaoguang) [inline content/docs/dev/concepts/explore-tikv-features/backup-restore.md:32]:
```suggestion
TiKV-BR, TiKV nodes, and the backup storage system should provide network bandwidth that is greater than the backup speed. If the target cluster is particularly large, the backup and restoration speed highly depends on the bandwidth of storage system bandwidth.  
```

Comment (haojinming) [reply content/docs/dev/concepts/explore-tikv-features/backup-restore.md]: Addressed and apply some suggestions from google docs, PTAL, thanks.

**Review by zeminzhou — APPROVED** [review]
LGTM~

**Pull request merged**